pennylane/operation.py

```diff
@@ -582,21 +582,79 @@ def decompose(self):
             return self.decomposition(wires=self.wires)
         return self.decomposition(*self.parameters, wires=self.wires)
 
+    @staticmethod
+    def compute_diagonalizing_gates(*params, wires, **hyperparams):
+        r"""Defines a partial representation of this operator via
+        its eigendecomposition.
+
+        Given the eigendecomposition `O = U \Sigma U^{\dagger}` where
```

Comment:
```suggestion
        Given the eigendecomposition :math:`O = U \Sigma U^{\dagger}` where
```

Reply:
I think it is the same :) 

Calling the operator `O` is more general, because it does not only apply to gates. However, the right and left matrices are guaranteed to be unitary (that's what an eigenvalue decomp is), so I call them `U`. In fact, this was from the previous docstring.

Or am I misunderstanding?

Reply:
sorry I shouldn't have changed the symbols 😆 Was too late when I wrote that. I meant more the `^{-1}` as opposed to the dagger --- wouldn't the dagger only hold if `O` is Hermitian?

Reply:
Because only hermiticity (or actually, more strictly real eigenspectrum) results in `U` being unitary?